### ISSUE TITLE
In-place/canary upgrade 1.6->1.7 failure on k8s 1.16

### ISSUE THREAD

**Issue by elfinhe** [istio/istio#26280]
**Bug description**
On k8s 1.16, Install Istio 1.6 first, then install Istio 1.7 for in-place upgrade. The 1.6 webhook will have a similar issue like: https://github.com/istio/istio/issues/26186. seems like we need to apply https://github.com/istio/istio/pull/26187 to release 1.6 as well to support inplace upgrade.

Background: https://github.com/kubernetes/kubernetes/issues/85681
Webhook v1beta1 -> v1 PR: https://github.com/istio/istio/pull/24723

```
$ ./istioctl install -f my-istio-operator.yaml
✔ Istio core installed                                        
2020-08-07T17:33:53.283001Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T17:33:53.476295Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T17:33:53.683472Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T17:33:53.890488Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T17:33:54.127200Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T17:33:54.320725Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
```
Same for canary upgrade:
```
$ ./istioctl install  -f my-istio-operator.yaml  --set revision=canary
✔ Istio core installed                                                                                                                                                 
2020-08-07T18:42:14.472831Z     error   installer       failed to create "EnvoyFilter/istio-system/metadata-exchange-1.6-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T18:42:14.595879Z     error   installer       failed to create "EnvoyFilter/istio-system/metadata-exchange-1.7-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T18:42:14.730029Z     error   installer       failed to create "EnvoyFilter/istio-system/stackdriver-filter-1.6-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T18:42:14.861414Z     error   installer       failed to create "EnvoyFilter/istio-system/stackdriver-filter-1.7-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T18:42:14.987809Z     error   installer       failed to create "EnvoyFilter/istio-system/tcp-metadata-exchange-1.6-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-07T18:42:15.116263Z     error   installer       failed to create "EnvoyFilter/istio-system/tcp-metadata-exchange-1.7-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
✘ Istiod encountered an error: failed to create "EnvoyFilter/istio-system/metadata-exchange-1.6-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
failed to create "EnvoyFilter/istio-system/metadata-exchange-1.7-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
failed to create "EnvoyFilter/istio-system/stackdriver-filter-1.6-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
failed to create "EnvoyFilter/istio-system/stackdriver-filter-1.7-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
failed to create "EnvoyFilter/istio-system/tcp-metadata-exchange-1.6-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
failed to create "EnvoyFilter/istio-system/tcp-metadata-exchange-1.7-canary": Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
```

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by elfinhe** [comment]
@howardjohn @tariq1890 ptal

**Comment by howardjohn** [comment]
We should not require a latest 1.6 to upgrade to 1.7 if possible. May need to revert the upgrade in 1.7

**Comment by elfinhe** [comment]
Yep, it would be better to have both v1&v1beta1 runtime scheme in 1.7, and update MutatingWebhook apiVersion to v1 in 1.8.

**Comment by tariq1890** [comment]
Reverting makes sense. So revert #24723 but retain #26187 , yes?

**Comment by howardjohn** [comment]
Yeah I think that makes sense to me?

**Comment by tariq1890** [comment]
Alrighty

**Comment by elfinhe** [comment]
Thanks Tariq

**Comment by elfinhe** [comment]
After some tests, I think we need to add `admissionReviewVersions: ["v1", "v1beta1"]` into both 1.7 and master branch to be able to support upgrade from 1.7 -> 1.8.

The revert PR https://github.com/istio/istio/pull/26285 has solved the issue in the 1.6->1.7 upgrade. But 1.7 cannot be upgraded to the planed 1.8 seamlessly without `admissionReviewVersions: ["v1", "v1beta1"]` line in istiod-istio-system ValidatingWebhookConfiguration.
